build/scripts/version-update/main.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var releaseType string
+
+func init() {
+	flag.StringVar(&releaseType, "type", "", "Specify the release type ('before' or 'after')")
+}
+
+func main() {
+	flag.Parse()
+
+	if len(os.Args) < 3 {
+		log.Fatalf("Please provide the release type ('before' or 'after') and the initial version as command-line arguments")
+		return
+	}
+
+	releaseType := os.Args[1]
```

Comment:
The way you are parsing this right now seems to ignore the flags - instead it looks like you would execute it via `version-update before 1.2.3`, vs `version-update --type=before 1.2.3`.

Since you are using the `flag` package, you should trust that the `flag` package can parse. For non-flag arguments, you should use [flag.Args](https://pkg.go.dev/flag#Args) .. so line 23 should be `if len(flag.Args()) < 1`, for instance.

`releaseType` is already initialized in `init()` above, so you don't need this line at all, and the line below should just be `initialVersion := flag.Args()[0]`.

Reply:
I have updated the code and passed the --release-stage=before flag from the command line.